### PR TITLE
Implement `GT_AND_NOT` for ARM/ARM64

### PR DESCRIPTION
**Pull request by SingleAccretion**
This is a simple implementation of the operator, targeting AArch, but with a structure explicit in its intent to expand this optimization to XArch as well, in the future.

Diffs: [`win-arm64`](https://github.com/SingleAccretion/diffs-repository/blob/main/runtime-59881/win-arm64.md), [`linux-arm`](https://github.com/SingleAccretion/diffs-repository/blob/main/runtime-59881/linux-arm.md).

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/jit-contrib, @tannergooding

**Comment by SingleAccretion** [comment]
Some conflicts need to be fixed.

File: src/coreclr/jit/lower.cpp [hunk 1]
```diff
@@ -5098,6 +5097,55 @@ GenTree* Lowering::LowerAdd(GenTreeOp* node)
...
+    // TODO-CQ-XArch: support BMI2 "andn" in codegen and condition
+    // this logic on the support for the instruction set on XArch.
+    CLANG_FORMAT_COMMENT_ANCHOR;
+
+#ifdef TARGET_ARMARCH
+    if (comp->opts.OptimizationEnabled() && node->OperIs(GT_AND))
+    {
+        GenTree* opNode  = nullptr;
+        GenTree* notNode = nullptr;
+        if (node->gtGetOp1()->OperIs(GT_NOT))
```
Comment (echesakovMSFT) [inline src/coreclr/jit/lower.cpp:5123]: Can both `op1` and `op2` be `GT_NOT`?

**Review by echesakov — APPROVED** [review]
Changes LGTM.
I have one follow-up question.

File: src/coreclr/jit/lower.cpp (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/jit/lower.cpp:5123]: Yes. I believe things should still work correctly in that case (because of the `else`).

Comment (echesakovMSFT) [reply src/coreclr/jit/lower.cpp:5123]: Okay, for some reason, I got confused originally - but I see now that shouldn't be an issue.

**Comment by echesakovMSFT** [comment]
It looks that a similar optimization can be done for `OR(op1, NOT(op2))` by utilizing `eon` instruction on Arm64.

**Pull request merged**